packages/viz/__tests__/ColorBinsStyle.spec.ts

```diff
@@ -1,10 +1,107 @@
-// eslint-disable-next-line import/no-unresolved
+import { Deck } from '@deck.gl/core';
 import { colorBinsStyle } from '../src/lib/style';
+import * as mapsResponse from './data-mocks/maps.number.json';
+import { CARTOSource } from '../src';
+import { hexToRgb } from '../src/lib/style/helpers/utils';
+import { CartoStylingError } from '../src/lib/errors/styling-error';
+
+const FIELD_NAME = 'pct_higher_ed';
+const mapStats = mapsResponse.metadata.layers[0].meta.stats;
+const sample = mapStats.sample.map(s => s.pct_higher_ed);
+const stats = {
+  name: FIELD_NAME,
+  ...mapStats.columns.pct_higher_ed,
+  sample
+};
+const getMetadata = jest.fn().mockImplementation(() => {
+  return {
+    geometryType: 'Polygon',
+    stats: [stats]
+  };
+});
+
+jest.mock('../src', () => ({
+  CARTOSource: jest.fn().mockImplementation(() => ({ getMetadata }))
+}));
+
+const styledLayer = {
+  getMapInstance: () => ({} as Deck),
+  source: new CARTOSource('table')
+};
 
 describe('ColorBinsStyle', () => {
   describe('Style creation', () => {
-    it('should create a sizeBinsStyle instance properly', () => {
+    it('should create a ColorBinsStyle instance properly', () => {
       expect(() => colorBinsStyle('attributeName')).not.toThrow();
     });
+
+    it('should always return a getFillColor function', () => {
+      const style = colorBinsStyle(FIELD_NAME);
+      const response = style.getLayerProps(styledLayer);
+      expect(response).toHaveProperty('getFillColor');
+      expect(response.getFillColor).toBeInstanceOf(Function);
+    });
+  });
+
+  describe('Parameters', () => {
+    it('should launch styling error when bins and breaks missmatch', () => {
+      const style = colorBinsStyle(FIELD_NAME, {
+        breaks: [20, 50],
+        bins: 1
+      });
+      expect(() => style.getLayerProps(styledLayer)).toThrow(CartoStylingError);
+    });
+
+    it('should launch styling error when breaks and palette missmatch', () => {
+      const style = colorBinsStyle(FIELD_NAME, {
+        breaks: [20, 50],
+        palette: ['#ff0', '#231']
+      });
+      expect(() => style.getLayerProps(styledLayer)).toThrow(CartoStylingError);
```

Comment:
The more I see the `getLayerProps` call, the more I feel we could rename it to be more clear:

How about?
```ts
  public getStylePropsFor(styledLayer?: StyledLayer) {
    if (typeof this._styleProperties === 'function') {
      if (styledLayer === undefined) {
        throw new CartoStylingError(
          'No layer instance when calling styles function',
          stylingErrorTypes.SOURCE_INSTANCE_MISSING
        );
      }

      return this._styleProperties(styledLayer);
    }

    return this._styleProperties;
  }
```